chain-connect/src/GalachainConnectTONClient.ts

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) Gala Games Inc. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { ChainCallDTO, ConstructorArgs, serialize, signatures } from "@gala-chain/api";
+import { Address, beginCell } from "@ton/core";
+import { getSecureRandomBytes, keyPairFromSeed, sign, signVerify } from "@ton/crypto";
+import { TonProofItemReplySuccess } from "@tonconnect/protocol";
+import { Account, TonConnect } from "@tonconnect/sdk";
+
+export class GalachainConnectTONClient {
+  #tonAddress: string;
+  #tonPublicKey: string;
+  #chainCodeUrl: string;
+  #accessToken: string | null = null;
+
+  get galachainAddress() {
+    return this.#tonAddress.replace("0x", "ton|");
+  }
+
+  get tonAddress() {
+    return this.#tonAddress;
+  }
+
+  get accessToken() {
+    return this.#accessToken;
+  }
+
+  constructor(chainCodeUrl: string, manifestUrl: string) {
+    // super();
+    const connector = new TonConnect({ manifestUrl });
+
+    this.#chainCodeUrl = chainCodeUrl;
+    this.initializeTONListeners(connector);
+  }
+
+  private initializeTONListeners(connector): void {
+    connector.onStatusChange((wallet) => {
+      if (!wallet) {
+        this.#accessToken = null;
+        return;
+      }
+
+      const tonProof = wallet.connectItems?.tonProof;
+
+      if (tonProof) {
+        if ("proof" in tonProof) {
+          this.#tonAddress = wallet.account.address;
+          this.#tonPublicKey = wallet.account.publicKey;
+          this.checkProof(tonProof.proof, wallet.account);
+          return;
+        }
+
+        console.error(tonProof.error);
+      }
+
+      if (!this.#accessToken) {
+        connector.disconnect();
+      }
+    });
+  }
+
+  private async checkProof(proof: TonProofItemReplySuccess["proof"], account: Account) {
```

Comment:
I don't think you need to check proof in any way. You need to connect to TON and get the signature to add it to DTO. Proof/signature verification happens chaincode-side.